src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -588,6 +590,10 @@ impl TargetInfo {
             .iter()
             .any(|sup| sup.as_str() == split.as_str())
     }
+
+    pub fn get_target_envs(&self) -> CargoResult<&BTreeMap<String, Option<OsString>>> {
+        return Ok(self.crate_type_process.get_envs());
+    }
 }
```

Comment:
This feels very brittle because any change to how we pass in environment variables automatically gets used as  fingerprinting, rather than intentionally choosing what we fingerprint.  Just to check to see if this is correct is requiring checking several other places.

In fact, I think this will cause every build to rebuild everything when using jobserver because I think we are capturing the env after that is configured and that env includes file descriptors.

Reply:
This was overlooking that we aren't fingerprinting all of it, only what the user requests.  So maybe its not all that of a problem?